packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+  public async doDelete(physicalId: string, resourceProperties: ISEPConfiguratorResourceProperties): Promise<void> {
+    // Nothing to do -- we don't modify anything.
+    return;
+  }
+
+  private implementsISEPConfiguratorResourceProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!this.implementsIConnectionOptions(value.connection)) { return false; }
+    if (!this.isValidSFRConfig(value.spotFleetRequestConfigurations)) { return false; }
+    if (!this.isValidSpotPluginConfig(value.spotPluginConfigurations)) { return false; }
+    return true;
+  }
+
+  private implementsIConnectionOptions(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!value.hostname || typeof(value.hostname) !== 'string') { return false; }
+    if (!value.port || typeof(value.port) !== 'string') { return false; }
+    const portNum = Number.parseInt(value.port, 10);
+    if (Number.isNaN(portNum) || portNum < 1 || portNum > 65535) { return false; }
+    if (!value.protocol || typeof(value.protocol) !== 'string') { return false; }
+    if (value.protocol !== 'HTTP' && value.protocol !== 'HTTPS') { return false; }
+    if (!this.isSecretArnOrUndefined(value.caCertificate)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.passphrase)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.pfxCertificate)) { return false; }
+    return true;
+  }
+
+  private isSecretArnOrUndefined(value: any): boolean {
+    if (value) {
+      if (typeof(value) !== 'string' || !isSecretArn(value)) { return false; }
+    }
+    return true;
+  }
+
+  private isValidSFRConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+    return true;
+  }
+
+  private isValidSpotPluginConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+
+    return true;
+  }
+
+  private async spotEventPluginRequests(connection: IConnectionOptions): Promise<EventPluginRequests> {
+    return new EventPluginRequests(new DeadlineClient({
+      host: connection.hostname,
+      port: Number.parseInt(connection.port, 10),
+      protocol: connection.protocol,
+      tls: {
+        ca: connection.caCertificate ? await this.readCertificateData(connection.caCertificate) : undefined,
+      },
+    }));
+  }
+
+  private convertToBoolean(input: string): boolean | undefined {
+    try {
+      return JSON.parse(input);
+    } catch(e) {
+      return undefined;
+    }
+  }
+
+  /**
+   * Passing spot fleet request configs into lambda converts all numbers and booleans into strings.
+   * This functions converts these values back to their original type.
+   */
+  private spotFleetRequestToString(spotFleetRequestConfigs: object): string {
```

Comment:
We pass here an object, not a string, because we want to make sure that all the lazy evaluations are working correctly and all the tokens are resolved.

Numeric and boolean properties are converted to strings during serialization, so we need this function to restore the original types, before we create a Spot Fleet request Configuration string. We do the same below in the `spotFleetPluginConfigsToArray` function.

Reply:
I agree.